lib/go/replication.pb.go

```diff
@@ -0,0 +1,1014 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// versions:
+// 	protoc-gen-go v1.25.0
+// 	protoc        v3.5.0
+// source: replication.proto
+
+package replication
+
+import (
+	proto "github.com/golang/protobuf/proto"
+	descriptor "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
+	reflect "reflect"
+	sync "sync"
+)
+
+const (
+	// Verify that this generated code is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(20 - protoimpl.MinVersion)
+	// Verify that runtime/protoimpl is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
+)
+
+// This is a compile-time assertion that a sufficiently up-to-date version
+// of the legacy proto package is being used.
+const _ = proto.ProtoPackageIsVersion4
+
+type EnableVolumeReplicationRequest struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	// The identifier for this volume, generated by the plugin during
+	// CreateVolume CSI RPC call.
+	// This field is REQUIRED.
+	// This field MUST contain enough information to uniquely identify
+	// this specific volume vs all other volumes supported by this plugin.
+	// This field SHALL be used by the CO in subsequent calls to refer to
+	// this volume.
+	VolumeId string `protobuf:"bytes,1,opt,name=volume_id,json=volumeId,proto3" json:"volume_id,omitempty"`
```

Comment:
I guess this `VolumeId` should not be `omitempty`. We do need to enforce proto here to make this mandatory field in all requests

Reply:
This is auto-generated file. we should not modify it manually. the driver should return an error if the VolumeID is empty. I will add the error codes in the next PR

Reply:
I also came across the reason why they removed it https://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3